woocommerce/woocommerce-gutenberg-products-block
# Storybook and TS migration of some cart checkout components

This PR adds stories for a few components related to the cart and checkout and migrates them to TypeScript.

The following components are within the scope of this PR:

* `totals/Coupon`
* `totals/Discount`
* `totals/FooterItem`
* `ProductName`

Fixes #5311

### Testing

### Manual Testing

How to test the changes in this Pull Request:

1. Run `npm run storybook`
2. Explore the [cart-checkout category stories](http://localhost:6006/?path=/story/woocommerce-blocks-base-components-cart-checkout-totals-coupon--default). Upon opening the category, you should see a sub-category called `totals` and overall the four components mentioned above should be available.

Specifically, these stories

* `totals/Coupon`:
    *  A Default story where, when clicking `Apply`, a loading action is simulated.
    * A story showing the loading state
    * A story showing the error state
* `totals/Discount`:
    *  A Default story where you can simulate the deletion of the coupon code by clicking on the `X`. It will simulate the loading state, an `action` will appear in the `Actions` tab, and after that the coupon will be removed and the discount total will be updated.
    * A story showing the loading state
* `totals/FooterItem`:
   * A default story
   * Story showing the item including taxes
* `ProductName`:
   * A default story with the link
   * A story showing the disabled state.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
